cassandra-executor/src/main/java/io/mesosphere/mesos/frameworks/cassandra/ObjectFactory.java

```diff
@@ -0,0 +1,160 @@
+package io.mesosphere.mesos.frameworks.cassandra;
+
+import com.fasterxml.jackson.dataformat.yaml.snakeyaml.Yaml;
+import com.google.common.base.Joiner;
+import com.google.common.io.Files;
+import io.mesosphere.mesos.frameworks.cassandra.jmx.JmxConnect;
+import io.mesosphere.mesos.frameworks.cassandra.jmx.ProdJmxConnect;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.Marker;
+
+import java.io.*;
+import java.nio.charset.Charset;
+import java.util.List;
+import java.util.Map;
+
+public interface ObjectFactory {
+
+    JmxConnect newJmxConnect(CassandraFrameworkProtos.JmxConnect jmx);
+
+    WrappedProcess launchCassandraNodeTask(Marker taskIdMarker, CassandraFrameworkProtos.CassandraServerRunTask cassandraServerRunTask) throws IOException;
+
+    public interface WrappedProcess {
+        void destroy();
+
+        int exitValue();
```

Comment:
please define custom runtime exception and declare as a throws on this method. Then in the implementation catch the process exception and re-throw as cause of custom execption.